pkg/provider/aws.go

```diff
@@ -23,8 +25,9 @@ type AWSProvider struct {
 	project       string
 	bucket        string
 	Reg           string `survey:"region"`
-	storageClient *s3.S3
+	storageClient *s3.Client
 	writer        manifest.Writer
+	goContext     context.Context
```

Comment:
Oh I do think we can recreate these in individual funcs instead of storing in the provider object, which can be clunky in its own right.

Reply:
Without doing this, `mkBucket()` and `Decommission()` would have an empty context rather than the one used while configuring the AWS client. Although to pass the context around properly I think it should probably be a pointer. It didn’t seem to matter when I was just using `context.TODO()` everywhere, but never know when it might come back to cause problems.

Reply:
why would it be bad to just use `context.Background()` or `context.TODO()` there?